bugs/gh-204.txt

```diff
@@ -0,0 +1,1151 @@
+=== Begin SRU Template ===
+[Impact]
+When an user is setting a new random password, we are now using the SystemRandom class. This class
+generates random numbers from sources provided by the operating system. Therefore, the password
+should be harder to crack than ones originally created.
+
+[Test Case]
+
+```
+#!/bin/sh
+set -x
+# Manually deploy on a lxc container
+
+cat > config.yaml <<EOF
+#cloud-config
+users:
+  - name: testcloudinit
+    sudo: false
+chpasswd:
+  expire: false
+  list:
+    - testcloudinit:RANDOM
+EOF
+
+
+get_boot_load_time() {
+    name=$1
+    lxc exec $name -- systemd-analyze
+    lxc exec $name -- systemd-analyze blame
+}
```

Comment:
Please also run cloud-init analyze blame as it will measure more discrete components of cloud-init to see if set_passwords modules runs an inordinate amount of time.